gridstatus/spp.py

```diff
@@ -213,17 +217,59 @@ def get_solar_and_wind_forecast_short_term(self, date, verbose=False):
         Returns:
             pd.DataFrame: forecast as dataframe.
         """
-        # The SHORT_TERM forecast is delayed up to 2 minutes.
+        # The short_term forecast is delayed up to 2 minutes.
         buffer_minutes = 2
 
+        now = self.now() - pd.Timedelta(minutes=buffer_minutes)
+
         if date in ["latest", "today"]:
-            date = pd.Timestamp.now(tz=self.default_timezone) - pd.Timedelta(
-                minutes=buffer_minutes,
-            )
+            date = now
 
-        return self._get_solar_and_wind_forecast_short_term(date, verbose=verbose)
+        date = utils._handle_date(date, self.default_timezone)
```

Comment:
the point of the `@support_date_range` decorator is to handle all the date logic before it gets to this function. 

let's not worry about adding more in this PR unless we are prepared to add it everywhere. 

for, example im not even sure we want to do this `date = min(now, date)`.

Reply:
Without this logic, the decorator will call this method for all 288 daily intervals, regardless of the current time. This results in a flood of error messages because, unless you run this late in the day, most of the calls fail. That's how I originally had the logic, but it's slow and a poor user experience.

Reply:
I can change it back to that though if we're fine with the error messages.

Reply:
i see. that make sense. if we don't want that behavior, let's just disable being able to use `"today"` or `"latest"`. user must specify a single absolute time or a range.  we should just support what the SPP api makes this easy 

I'd rather do that then add more logic here we need to maintain.